platform/ios/src/MGLMapView.mm

```diff
@@ -3053,6 +3055,10 @@ - (void)unrotateIfNeededAnimated:(BOOL)animated
             [self resetNorthAnimated:NO];
         }
     }
+    else if (self.compassView.alpha > 0 && (self.direction < snapTolerance || self.direction > 360 - snapTolerance))
```

Comment:
What if the developer hides the compass view? Shouldn’t the map view snap to north regardless?